File: API/Entities/RecurringTransaction.cs
namespace API.Entities;

public enum Frequency
{
    Daily,
    Weekly,
    Monthly,
    Yearly
}

public class RecurringTransaction
{
    public Guid Id { get; set; }
    public required string Label { get; set; }
    public required decimal Amount { get; set; }
    public required TransactionType Type { get; set; }
    public Frequency Frequency { get; set; }

    /// Prochaine échéance à générer. Avancée par RecurringTransactionGenerationJob
    /// après chaque transaction générée (§ Génération automatique).
    public DateOnly NextDueDate { get; set; }

    /// Met en pause sans perdre la configuration (montant, catégorie,
    /// fréquence) — le job ignore toute transaction récurrente Active == false.
    public bool Active { get; set; } = true;

    public Guid CategoryId { get; set; }
    public Category Category { get; set; } = null!;

    public Guid UserId { get; set; }
    public AppUser User { get; set; } = null!;

    /// Transactions générées par ce modèle. Détachées (RecurringTransactionId à
    /// null), jamais supprimées, si ce modèle est supprimé.
    public ICollection<Transaction> Transactions { get; set; } = [];
}

File: client/src/types/recurring-transaction.ts
import { TransactionType } from './transaction';

export type Frequency = 'Daily' | 'Weekly' | 'Monthly' | 'Yearly';

export type RecurringTransaction = {
  id: string;
  label: string;
  amount: number;
  type: TransactionType;
  frequency: Frequency;
  /** "2026-08-16" */
  nextDueDate: string;
  active: boolean;
  categoryId: string;
  categoryName: string;
  categoryTranslationKey?: string | null;
  categoryIcon?: string | null;
  categoryColor?: string | null;
};

export type CreateRecurringTransactionDto = {
  label: string;
  amount: number;
  type: TransactionType;
  frequency: Frequency;
  nextDueDate: string;
  active: boolean;
  categoryId: string;
};

export type UpdateRecurringTransactionDto = CreateRecurringTransactionDto;
